packages/runtime-vapor/src/block.ts

```diff
@@ -67,16 +67,37 @@ export class DynamicFragment extends VaporFragment {
 
     if (this.fallback && !isValidBlock(this.nodes)) {
       parent && remove(this.nodes, parent)
-      this.nodes =
-        (this.scope || (this.scope = new EffectScope())).run(this.fallback) ||
-        []
+      // handle nested dynamic fragment
+      if (isFragment(this.nodes)) {
+        renderFallback(this.nodes, this.fallback, key)
```

Comment:
I believe this would cause other `VaporFragment` types that are not `DynamicFragment` (like `createFor` with an empty list as the source) to not render the fallback as it would bypass any node assignment